core/templates/components/state-directives/outcome-editor/outcome-destination-editor.component.ts

```diff
@@ -76,8 +76,10 @@ export class OutcomeDestinationEditorComponent implements OnInit {
   }
 
   updateChanges($event: string): void {
-    this.outcomeNewStateName = $event;
-    this.getChanges.emit();
+    if ($event !== '') {
```

Comment:
This seems odd, why would $event even be ''? Shouldn't we try to stop $event firing with an empty string in the first place?

Reply:
This comes from the input field in the UI. It can fire with `''`.